### PR TITLE
add a notebook for MLP classifier

### PR DESCRIPTION
**Pull request by abcdefgs0324**
add a notebook for MLP classifier as a start point
- Feed embeddings from language model into downstream MLP
- Write a MLP classifier
- Use `kubernetes/kubernetes` to show the auc performance

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/issue-label-bot/1)

### CODE REVIEW

**Comment by hamelsmu** [comment]
Great work!  One thing you should add is calculate probability thresholds that achieve the desired precision / recall for each label.  

Eventually, we want the user to be able to choose the precision / recall trade off, however to keep things simple in the beginning I suggest using this heuristic:

- Calculate the probability threshold for precision at 70% for each label.  You will need to save this for inference time.
- Report the recall at this precision 
- Exclude the prediction of any labels that do not achieve at least 50% recall.  You will have to also save this exclusion list such that you ignore these labels at prediction time because model is “not good enough” for these labels.

Probably good to make precision and recall parameters we can tweak.  Let me know if this makes sense.

**Comment by abcdefgs0324** [comment]
Good suggestion! I will do it.

One quick question, I was confused about your third bullet point. Will I save an exclusion list for some labels such that I ALWAYS ignore these labels in the list for all issues at prediction time? Or, do you mean I should possibly predict these labels for each issue but if the highest probability is not enough, I ignore to label it?

To me, the second one sounds better. In my imagination, users will choose labels which need to be predicted. And, while our bot works, we automatically label those with enough confidence (i.e., the probability of the label is larger than a threshold). But if for an issue, the highest probability of labels is lower than the threshold, our bot will not do any labeling.

To sum up, users will decide two things while installing our bot: first, the labels that should be tagged. Second, the threshold for the confidence to tag a label. Does that sound good?

I will include the precision/recall calculation in the notebook. And probably we will have some new ideas when we finish the first version of repo-specific bot. Thanks.

**Comment by hamelsmu** [comment]
@abcdefgs0324 (that is a really interesting GitHub handle, by the way) some clarification on my thoughts:

The only issue with the user choosing the probability threshold (either a global one or separate one for each label) is this will require a user interface so that the user can make an **informed** choice based on the recall/precision impact of choosing such a threshold (setting a threshold without this additional context requires the user to guess in the dark and is very frustrating).  This will require one of us to make a user interface with a slider for each label that displays the precision and recall for a particular choice or something else that allows the user to calibrate their desired precision/recall or simply ignore the label.  What I was proposing, is assuming we are not making this user interface in version 1 you set the precision / recall thresholds for the user and ALWAYS ignore labels that do not meet these criteria.   The reason for this is the user community is very sensitive to too many false positives, so if the model is incorrectly labeling a particular label 45% of the time even though the threshold is 0.5 this would be very bad.    

Note that in addition to this we can have a YAML file that allows the user to manually list the labels they would like to ignore including regex patterns of labels they would like to ignore.  Let me know if you have any comments or concerns on this aspect.  

@jlewi  let us know if you have any preferences

**Comment by abcdefgs0324** [comment]
@hamelsmu  Okay, I got your point and I agree with you. Thanks for the clarification!

**Comment by jlewi** [comment]
My only suggestion would be to decouple the problem of what gets specified in the YAML file from how users figure out what the correct values are to put in the YAML file.

Ideally the values in the YAML file would map to things the user cares about. I think this is what @hamelsmu is suggesting by possibly making the values the precision and recall rates as opposed to the confidence.

**Comment by jlewi** [comment]
@hamelsmu if you want to approve this PR add a comment "/approve" and "/lgtm" then our bots will be able to merge it.

**Comment by hamelsmu** [comment]
@abcdefgs0324 you might want to consider moving this notebook into the Issue-Label-Bot-v2 folder
